### DISCUSSION TITLE
Subversion/git-github troubles

### DISCUSSION THREAD

**Discussion by bozimmerman** [community/community#21681] [New to GitHub]
<p>Hello,</p>
<p>I use github as a final publication point for a subversion repository.  In the past, this has gone well.  I’m familiar with how to create and maintain a git repo that mirrors the changes checked into subversion. </p>
<p>My troubles started, however, when the subversion repo had to be restored from backups, making the <a href="http://github.com" rel="nofollow noopener">github.com</a> repo one (1) check-in ahead of the existing subversion repo.  </p>
<p>Since then, I’ve had to add a step where I update my local git repo from subversion, then merge changes from github, and finally push all changes to github.  This is not going to work forever.  I see two solutions, neither of which I know how to do:</p>
<ol>
<li>
<p>Figure out how to get the change in github merged back down to the subversion repo in a way that github will recognize.</p>
</li>
<li>
<p>Rebuild my github repo from scratch using a prestine version from the head of subversion, without losing all the meta-data, follows, etc attached to my existing github repo.</p>
</li>
</ol>
<p>I know, it’s complicated.  Any help is appreciated.</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>Since you’re already pulling in changes from subversion and the GitHub-based git repository, then you should be able to do the following:</p>
<ol>
<li>Bring in the latest changes from subversion to your local git repository</li>
<li>Use a “force push” to tell the GitHub-based repository to use exactly what you have locally, thereby dropping the “one commit ahead”</li>
</ol>
<p>You can do this by using <code>git push --force</code> instead of <code>git push</code>.</p>
<p>?  <strong>WARNING: </strong> A force push is a potentially destructive action.</p>
<p>But you want to destroy the one extra commit, so it s</p>

**Comment by bozimmerman** [comment]
<p>Thank you very much lee-dohm!</p>
